website/docs/help-and-support/troubleshooting-guide/action-errors/datasource-errors.md

```diff
@@ -66,14 +66,41 @@ At least one of the mandatory fields in the plugin's datasource creation form is
 
 This message indicates that one of the mandatory fields, for example, `Access Key`, has been left empty in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
 
-This error can be fixed by filling the mentioned mandatory field(s) in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
+This error can be fixed by filling the mentioned mandatory fields in the [datasource creation form](../../../core-concepts/connecting-to-data-sources/connecting-to-databases.md).
 
-### Cannot Delete Datasource Error
+### Can't delete datasource error
 
 ```
 Cannot delete datasource since it has 1 action(s) using it.
 ```
 
-This message indicates that the [datasource](../../../reference/datasources/) attempting to be deleted has some query action configured on it.
+This message indicates that the [datasource](/reference/datasources/) attempting to be deleted has some query action configured on it.
 
-This error can be fixed by deleting any queries dependent on this [datasource](../../../reference/datasources/) before attempting to delete the [datasource](../../../reference/datasources/).
+This error can be fixed by deleting any queries dependent on this [datasource](/reference/datasources/) before attempting to delete the [datasource](../../../reference/datasources/).
+
+### Error connecting to local DB or API
+
+If you are trying to connect to a local MySQL database from Appsmith and see an error message like:
```

Comment:
@sumitsum - Could you please confirm if the error message is specific to MySQL database and only for self-hosted instance?

Reply:
+1

Reply:
I think there is some confusion here. Can we get on a quick call to resolve this ?

Reply:
Although, the error example quoted below is specific to MySQL (because of port no 3306) similar error can be seen for other plugins as well. IMO, we shouldn't use the `MySQL` plugin word for this error and show it as a general error.

Reply:
@harshilp24 can you please change this line above `local MySQL database` to `local database` ?

Reply:
done